pulsar-client-tools/pom.xml

```diff
@@ -63,6 +63,12 @@
 		<dependency>
 			<groupId>org.asynchttpclient</groupId>
 			<artifactId>async-http-client</artifactId>
+			<exclusions>
+				<exclusion>
+					<groupId>io.netty</groupId>
+					<artifactId>netty-buffer</artifactId>
```

Comment:
Yes, actually entry point of the issue was conflicting ```io.netty.buffer.PoolThreadCache``` under ```netty-buffer``` and ```netty-all``` and removing of ```netty-buffer``` solves it. However, we can also remove all netty dependencies from ```async-http-client``` as well. I have added change to parent pom as ```pulsar-client``` and ```pulsar-client-tool``` use them. 

```
java.lang.NoSuchMethodError: io.netty.util.internal.MathUtil.safeFindNextPositivePowerOfTwo(I)I
        at io.netty.buffer.PoolThreadCache$MemoryRegionCache.<init>(PoolThreadCache.java:372) ~[netty-buffer-4.0.42.Final.jar:4.0.42.Final]
        at io.netty.buffer.PoolThreadCache$SubPageMemoryRegionCache.<init>(PoolThreadCache.java:340) ~[netty-buffer-4.0.42.Final.jar:4.0.42
.Final]
        at io.netty.buffer.PoolThreadCache.createSubPageCaches(PoolThreadCache.java:135) ~[netty-buffer-4.0.42.Final.jar:4.0.42.Final]
        at io.netty.buffer.PoolThreadCache.<init>(PoolThreadCache.java:86) ~[netty-buffer-4.0.42.Final.jar:4.0.42.Final]
        at io.netty.buffer.PooledByteBufAllocator$PoolThreadLocalCache.initialValue(PooledByteBufAllocator.java:361) ~[netty-buffer-4.0.42.Final.jar:4.0.42.Final]
        at io.netty.buffer.PooledByteBufAllocator$PoolThreadLocalCache.initialValue(PooledByteBufAllocator.java:354) ~[netty-buffer-4.0.42.Final.jar:4.0.42.Final]
        at io.netty.util.concurrent.FastThreadLocal.initialize(FastThreadLocal.java:155) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.util.concurrent.FastThreadLocal.get(FastThreadLocal.java:149) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.util.concurrent.FastThreadLocal.get(FastThreadLocal.java:135) ~[netty-all-4.0.39.Final.jar:4.0.39.Final]
        at io.netty.buffer.PooledByteBufAllocator.newHeapBuffer(PooledByteBufAllocator.java:251) ~[netty-buffer-4.0.42.Final.jar:4.0.42.Final]
        at io.netty.buffer.AbstractByteBufAllocator.heapBuffer(AbstractByteBufAllocator.java:158) ~[netty-buffer-4.0.42.Final.jar:4.0.42.Final]
        at io.netty.buffer.AbstractByteBufAllocator.heapBuffer(AbstractByteBufAllocator.java:149) ~[netty-buffer-4.0.42.Final.jar:4.0.42.Final]
```